### PR TITLE
SLVSCODE-245 command for compilation database setup

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -527,6 +531,42 @@ function clearLocations() {
...
 }
 
+interface IndexQP extends VSCode.QuickPickItem {
+  index: number;
+}
+
+async function ShowCDOptions(paths: VSCode.Uri[]) {
+  const setMessage = `Analysis configured. Compilation database path is set to: `;
+  if (paths.length === 1) {
+    VSCode.window.showInformationMessage(setMessage + paths[0].fsPath);
```
Comment (henryju) [inline src/extension.ts:541]: I think in TypeScript it is better to use string templates than concatenation

File: src/extension.ts [hunk 2]
```diff
@@ -527,6 +531,42 @@ function clearLocations() {
...
+      VSCode.ConfigurationTarget.Workspace);
+  }
+  return Promise.resolve();
+}
+
+async function configureCompilationDatabase() {
+  const paths = (await VSCode.workspace.findFiles(`**/compile_commands.json`))
+      .filter((path) => FS.existsSync(path.fsPath));
+  if (paths.length === 0) {
+    VSCode.window.showWarningMessage(`No compilation database was found in the workspace`);
```
Comment (henryju) [inline src/extension.ts:564]: Shouldn't be plural form with 'no'? `No compilation databases were found....`

**Comment by henryju** [comment]
Maybe @jblievremont you could have a look, as I'm not very experimented with TypeScript.

File: src/extension.ts [hunk 3]
```diff
@@ -527,6 +531,42 @@ function clearLocations() {
   issueLocationsView.message = null;
 }
 
+interface IndexQP extends VSCode.QuickPickItem {
+  index: number;
+}
+
+async function ShowCDOptions(paths: VSCode.Uri[]) {
```
Comment (jblievremont) [inline src/extension.ts:538]:
```suggestion
async function showCompilationDatabaseOptions(paths: VSCode.Uri[]) {
```
Function name should start with a lower case `s`, and I believe that `CD` is a bit obscure to someone reading the code.

File: src/extension.ts [hunk 4]
```diff
@@ -527,6 +531,42 @@ function clearLocations() {
...
+    VSCode.window.showInformationMessage(setMessage + paths[0].fsPath);
+    return VSCode.workspace.getConfiguration().update(PATH_TO_COMPILE_COMMANDS, paths[0].fsPath,
+      VSCode.ConfigurationTarget.Workspace);
+  }
+  const items: IndexQP[] = [];
+  for (let i = 0; i < paths.length; i++) {
+    items.push({ label: paths[i].fsPath, description: ``, index: i });
+  }
+  const options: VSCode.QuickPickOptions = {};
+  options.placeHolder = `Pick a compilation database`;
```
Comment (jblievremont) [inline src/extension.ts:550]:
```suggestion
  const options = { placeHolder: 'Pick a compilation database' };
```

**Review by jblievremont — COMMENTED** [review]
A few style issues but LGTM on the functional side.

**Review by henryju — APPROVED** [review]

**Review by jblievremont — APPROVED** [review]

**Pull request merged**